File: src/main/java/main/utils/pagination/pages/MenuPage.java
package main.utils.pagination.pages;

import lombok.Getter;
import main.utils.component.interactions.selectionmenu.StringSelectMenuOption;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.util.ArrayList;
import java.util.List;

public class MenuPage extends MessagePage {
    @Getter
    private final List<StringSelectMenuOption> options = new ArrayList<>();

    public void addOption(StringSelectMenuOption option) {
        if (options.size() >= 25)
            throw new IllegalArgumentException("You cannot add any more options to the page!");

        options.add(option);
    }

    public void removeOption(int i) {
        options.remove(i);
    }

    public StringSelectMenuOption getOption(int i) {
        return options.get(i);
    }

    public List<String> toStringList() {
        final List<String> ret = new ArrayList<>();
        for (StringSelectMenuOption option : options) {
            if (option.toString().contains("Next Page") ||
                    option.toString().contains("Previous Page")
            ) continue;

            ret.add(option.toString());
        }

        return ret;
    }

    @Override
    public MessageEmbed getEmbed() {
        return null;
    }
}

File: src/main/kotlin/main/utils/pagination/pages/MenuPage.kt
package main.utils.pagination.pages

import main.utils.component.interactions.selectionmenu.StringSelectMenuOption
import net.dv8tion.jda.api.entities.MessageEmbed

class MenuPage : MessagePage {
    private val options = mutableListOf<StringSelectMenuOption>()

    override suspend fun getEmbed(): MessageEmbed? {
        return null
    }

    fun getOptions(): List<StringSelectMenuOption> =
        options.toList()

    fun addOption(option: StringSelectMenuOption) {
        check(options.size < 25) { "You cannot add anymore options to the page!" }
        options.add(option)
    }

    fun removeOption(index: Int) {
        options.removeAt(index)
    }

    fun getOption(index: Int): StringSelectMenuOption =
        options[index]

    fun toStringList(): List<String> =
        options.map { option ->
            if (!option.toString().contains("Next Page") && !option.toString().contains("Previous Page"))
                return@map option.toString()
            else return@map null
        }.filterNotNull()
}